### PR TITLE
Clippy lints and no more #![allow] in src/

### PR DESCRIPTION
**Pull request by itytophile**
Hi, it's me again. This is a PR to improve the code with clippy lints (with the help of rust-analyzer). I've also removed the allow attributes. These were used to not drop some tasks. The trick to avoid that is to use the [detach() method](https://docs.rs/bevy/0.6.0/bevy/tasks/struct.Task.html#method.detach). I've run the examples and they work as intended. I fixed the src/channel.rs file however there are some pinning in it and I'm not really used to that. Please watch out for potential mistakes.

### CODE REVIEW

File: examples/simple.rs [hunk 1]
```diff
@@ -53,11 +53,9 @@ fn startup(mut net: ResMut<NetworkResource>, args: Res<Args>) {
 }
 
 fn send_packets(mut net: ResMut<NetworkResource>, time: Res<Time>, args: Res<Args>) {
-    if !args.is_server {
-        if (time.seconds_since_startup() * 60.) as i64 % 60 == 0 {
-            info!("PING");
-            net.broadcast(Packet::from("PING"));
-        }
+    if !args.is_server && (time.seconds_since_startup() * 60.) as i64 % 60 == 0 {
```
Comment (smokku) [inline examples/simple.rs:56]: I don't like this change. These conditions are not related.

First there is a test for the context we are in `if !args.is_server`, then in the context there is another decision of what to do.

File: examples/simple.rs [hunk 2]
```diff
@@ -66,23 +64,20 @@ fn handle_packets(
     mut reader: EventReader<NetworkEvent>,
 ) {
     for event in reader.iter() {
-        match event {
```
Comment (smokku) [inline examples/simple.rs:69]: This code serves as an example how to approach reading events. Usually we are interested in processing all types of events, not just one specific (`if let Specific(`), so rewriting it as `if let` counters educational value.

File: src/channels.rs [hunk 3]
```diff
@@ -60,9 +41,7 @@ impl Runtime for TaskPoolRuntime {
     type Sleep = Pin<Box<dyn Future<Output = ()> + Send>>;
 
     fn spawn<F: Future<Output = ()> + Send + 'static>(&self, f: F) {
-        let task = self.pool.spawn(Box::pin(f));
-        #[cfg(not(target_arch = "wasm32"))]
-        self.tasks.lock().unwrap().push(task);
+        self.pool.spawn(Box::pin(f)).detach();
```
Comment (smokku) [inline src/channels.rs:44]: Are you sure this will not cause a dangling thread if someone manually destroys Bevy App.

File: examples/simple.rs (hunk 1 above)
Comment (itytophile) [reply examples/simple.rs:56]: Ok, if this is really important we can disable the lint that changed that code. Or we can keep the modification and put a comment to tell the reader what he could do.

File: examples/simple.rs (hunk 2 above)
Comment (itytophile) [reply examples/simple.rs:69]: I see. This is the same problem as the one with the "compressed" if.

File: src/channels.rs (hunk 3 above)
Comment (itytophile) [reply src/channels.rs:44]: That's interesting. I have not thought about that. I can investigate and maybe find a better suited pattern that can please that condition and clippy.

Comment (smokku) [reply src/channels.rs:44]: Yeah. Before states were a thing, people used to shut down a "menu" bevy app and proceed to "gameplay" bevy app, configured differently.

Comment (itytophile) [reply src/channels.rs:44]: Hi, this is about this [piece of code](https://github.com/smokku/bevy_networking_turbulence/blob/3deab3ccad69a5d81294c6bdc605f578431ccc37/src/transport.rs#L340). I don't understand why you spawn a task and immediately drop it if the code is compiled to wasm.

Comment (smokku) [reply src/channels.rs:44]: On WASM tasks are spawned as `async` and are tracked by JS runtime. On native these are real threads and we need to keep a live object, because `drop` shuts down a thread.

**Comment by itytophile** [comment]
Unfortunately I've used the original trick to remove the detach calls, I didn't find another way. But for whatever reason, clippy is happy so I am as well.

**Comment by smokku** [comment]
LGTM

Did you check examples/simple and example/channels work with your changes?

**Comment by itytophile** [comment]
Yes, I did it works as expected. I run simple.rs as a wasm app too.

**Comment by smokku** [comment]
> Yes, I did it works as expected. I've run simple.rs as a wasm app too.

Awesome!

**Review by smokku — APPROVED** [review]

**Review by smokku — APPROVED** [review]

**Comment by smokku** [comment]
TY @itytophile

**Comment by itytophile** [comment]
You're welcome, thanks for merging :pray:
